processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -35,6 +37,8 @@ public TopNMapFn(
   {
     this.query = query;
     this.topNAlgorithm = topNAlgorithm;
+    this.dimension = query.getDimensionSpec().getDimension();
+    this.extractionFn = TopNQueryEngine.canApplyExtractionInPost(query) ? null : query.getDimensionSpec().getExtractionFn();
```

Comment:
This one and the other TopN need to stay to get the proper (optimized) topN behavior at all levels.